### DISCUSSION TITLE
Runtime execution?

### DISCUSSION THREAD

**Discussion by kissu** [unocss/unocss#80] [Q&A] [answered]
Thanks (again) for this super interesting project. ❤️ 

Hi, I was wondering if Uno is somehow able to generate the CSS at runtime looking at it's nature.
Meaning that an SPA could call an API during it's mounting process, get a list of rules from the user (looking like a `windi.config.js` file) and create only the needed CSS.
Or somehow customize a preset?

Otherwise, we'll fallback to a build but the idea was to have something dynamic from a tech-savy'ish user + not having to re-build the whole app every time.

**Comment by antfu** [comment, marked as answer]
Even in windi it's possible: https://github.com/antfu/windicss-runtime-dom (but ofc, uno will be much smaller and faster). It is currently not on my top priority but please feel free to give it a try.

**Reply by kissu** [reply]
Damn, perfect! 🙏🏼

**Reply by yankeeinlondon** [reply]
@kissu did you ever try this out? Any tricks you had to do to get it to work?

**Reply by antfu** [reply]
Here is a CodePen example: https://codepen.io/antfu/pen/eYGNXGo
